SSA/Projects/RISCV64/Semantics.lean

```diff
@@ -992,17 +992,19 @@ def ZBB_RTYPE_pure_RISCV_MINU_bv (rs2_val : BitVec 64) (rs1_val : BitVec 64) : B
 theorem ZBB_RTYPE_pure_RISCV_MINU_eq_ZBB_RTYPE_pure_RISCV_MINU_bv (rs2_val : BitVec 64) (rs1_val : BitVec 64) :
     ZBB_RTYPE_pure_RISCV_MINU rs2_val rs1_val = ZBB_RTYPE_pure_RISCV_MINU_bv rs2_val rs1_val := by
   unfold ZBB_RTYPE_pure_RISCV_MINU ZBB_RTYPE_pure_RISCV_MINU_bv
-  simp
-  sorry
+  simp only [min_def, ule_iff_toNat_le]
+  norm_cast
+  split_ifs <;> simp
 
 def ZBB_RTYPE_pure_RISCV_MAXU_bv (rs2_val : BitVec 64) (rs1_val : BitVec 64) : BitVec 64 :=
   BitVec.extractLsb' 0 64 (if BitVec.ult rs1_val rs2_val then rs2_val else rs1_val)
 
 theorem ZBB_RTYPE_pure_RISCV_MAXU_eq_ZBB_RTYPE_pure_RISCV_MAXU_bv (rs2_val : BitVec 64) (rs1_val : BitVec 64) :
     ZBB_RTYPE_pure_RISCV_MAXU rs2_val rs1_val =ZBB_RTYPE_pure_RISCV_MAXU_bv rs2_val rs1_val := by
   unfold ZBB_RTYPE_pure_RISCV_MAXU ZBB_RTYPE_pure_RISCV_MAXU_bv
-  simp
-  sorry
+  simp only [max_def, ult_iff_toNat_lt]
+  norm_cast
+  split_ifs <;> simp <;> bv_omega
```

Comment:
One last thing: having non-terminal `simp` makes our design brittle, and we should avoid that. The theorem that `simp` applies can change and include other theorems that might change the result of the rewrite and potentially make the subsequent tactic (in this case `bv_omega`) unable to close the goal. To avoid that, if the `simp` is not terminal in the proof, we want to specify _exactly_ which theorems it applies. You can see this by typing `simp?`: the infoview will show you the set applied and you'll have something like `simp [lemma1, lemma2, lemma3]` at the end. Because different branches of the if can use different theorems, you might need an specific branch for each of them (i.e., without `<;>`).